src/buffer/mutable.rs

```diff
@@ -93,6 +93,19 @@ impl<T: NativeType> MutableBuffer<T> {
         }
     }
 
+    /// Allocates a new [MutableBuffer] with `len` and capacity to be at least `len` where
+    /// all bytes are not initialized
+    #[inline]
+    pub fn from_len(len: usize) -> Self {
```

Comment:
This is `unsafe` as it allows consumers to read un-initialized memory. The way to achieve this behavior is to use `with_capacity()` followed by `set_len()` (followed by an operation that is guaranteed to not read any entry and write to all entries up to len).